catalog/recursion-to-filter.md

```diff
@@ -0,0 +1,87 @@
+# Recursion to Filter
+
+## Relevant To
+
+
+* Replace Loop with Collection Closure Method
+* Replace Temp with Chain
+
+## Relates To
+
+* Recursion to Map
+* Recursion to Reduce
+* Tail Call Optimise Recursion
+
+## Languages
+
+All
+
+## Motivation
+
+Selectively filter a list of items into a new, smaller list of items, based
+on defined criteria, in the form of a boolean statement or function.
+
+## Description
+
+The filter takes a function (called a predicate) that is passed each element from
```

Comment:
Maybe here we can say something about identifying the opportunity to refactor and the predicate to be extracted?